AK/SipHash.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+
+namespace AK {
+
+// Ported from the SipHash reference implementation, released to the public domain:
+// https://github.com/veorq/SipHash/blob/eee7d0d84dc7731df2359b243aa5e75d85f6eaef/siphash.c
+// The standard is SipHash-2-4, but we use 1-3 for a little more speed.
+// Cryptography should use 4-8 for conservative security.
+// Make sure to always use the same SipHash instance for a container so that the random key stays the same!
+template<size_t message_block_rounds = 1, size_t finalization_rounds = 3>
+struct SipHash {
+    unsigned operator()(u64 input);
+};
+
+template<size_t message_block_rounds, size_t finalization_rounds>
+u64 sip_hash_bytes(ReadonlyBytes input);
+
+static SipHash<1, 3> StandardSipHash;
+static SipHash<4, 8> SecureSipHash;
+
+inline unsigned standard_sip_hash(u64 input)
+{
+    return StandardSipHash(input);
+}
+
+inline unsigned secure_sip_hash(u64 input)
+{
+    return SecureSipHash(input);
+}
```

Comment:
It's nice that there's a way to invoke the hasher on a u64, but there's a lot of API weirdness here:
- What happens if I call this with a `u32`? is the result equivalent to `sip_hash_bytes({ &value, sizeof(u32) })`?
- Why does this return an `unsigned` when `sip_hash_bytes` returns a `u64`?

Reply:
1. No; this is a mild oversight but doesn't matter for any of our use cases. Of course this problem is caused by C++ and short of templates I don't know of a good solution.
2. Our existing hash functions use `unsigned`, including Traits. Changing that would be massive and highly invasive so I opted to not touch that here.